cfi_csrs.adoc

```diff
@@ -65,28 +56,32 @@ backward-edge and forward-edge CFI at S-mode and for each application at U-mode.
   {bits:  1, name: 'CBZE'},
   {bits: 51, name: 'WPRI'},
   {bits:  1, name: 'SFCFIE'},
-  {bits:  1, name: 'CFIE'},
+  {bits:  1, name: 'SBCFIE'},
   {bits:  1, name: 'HADE'},
   {bits:  1, name: 'PBMTE'},
   {bits:  1, name: 'STCE'},
 ], config:{lanes: 4, hspace:1024}}
 ....
 
-The `CFIE` field (bit 60) controls whether the Zicfisslp extension is available
-for use in VS and VU modes. When `menvcfg.CFIE` is 0, `henvcfg.CFIE` is
-read-only zero.
-
-When `henvcfg.CFIE` bit is 0, then at privilege modes VS and VU:
+Zicfiss extension introduces the `SBCFIE` field (bit 60) in `henvcfg`. When
+`SBCFIE` field is 1, the Zicfiss extension is active in VS-mode. When `SBCFIE`
+field is 0, the Zicfiss extension is not active in VS-mode and the following
+rules apply when `V=1`.
 
-* Attempts to access the `ssp` CSR result in an virtual-instruction exception.
-* Shadow stack instructions revert to the Zimop/Zcmop defined behavior.
-* The `lpad` instruction executes as a no-op.
-* The `UBCFIE`, `UFCFIE`, and `SPELP` fields in `sstatus` (really `vsstatus`) are
-  read-only zero.
+* Attempts to access the `ssp` CSR raise an illegal-instruction exception.
+* The 32-bit Zicfiss instructions revert to their Zimop defined behavior.
+* The 16-bit Zicfiss instructions revert to their Zcmop defined behavior.
 * The `pte.xwr=010b` encoding in VS-stage page tables is reserved.
+* The `sstatus.UBCFIE` field is read-only zero.
```

Comment:
These are rules apply for `V=1` where `sstatus` aliases to `vsstatus`. This also follows the modulation rule that in `sstatus` at `V=1` they are read-only zero.

Reply:
I don't think it makes sense to ascribe a storage property like "read-only zero" to an alias - we should describe it as an access property (sstatus.UBCFIE ignores writes and reads as zero, but vsstatus.UBCFIE retains values written) or a storage property on the actual underlying field (vsstatus.UBCFIE is read-only zero, even if V=0).

This may also apply in 2.1; we aren't clear if nonzero values can be written to vsstatus.UBCFIE and mstatus.UBCFIE in M-mode if menvcfg.SBCFIE=0.

Reply:
I added "(really `vsstatus.UBCFIE`)" to clarify. It is read-only zero when V=1. 
`menvcfg` only affects operations at lower privilege than M. For example, consider `menvcfg.STCE`. When `STCE` in `menvcfg` is zero, an attempt to access `stimecmp` or `vstimecmp` in a mode other than M-mode raises an illegal instruction exception.

Reply:
Good catch on the rules concerning `menvcfg` and `senvcfg`.  That means that both this and 2.1 must be access properties, since the setting of bits in menvcfg and senvcfg cannot affect the allowed values for `vsstatus.UBCFIE` or `mstatus.UBCFIE` when accessed in M-mode.